src/main/java/com/bobocode/bibernate/session/SessionImpl.java

```diff
@@ -28,15 +33,24 @@ public class SessionImpl implements Session {
 
     private final EntityPersister entityPersister;
 
-    public SessionImpl(EntityPersister entityPersister, Dialect dialect) {
+    private final PersistenceContext persistenceContext;
+
+    public SessionImpl(EntityPersister entityPersister, Dialect dialect, PersistenceContext persistenceContext) {
         this.dialect = dialect;
         this.entityPersister = entityPersister;
+        this.persistenceContext = persistenceContext;
     }
 
     @Override
     public <T> Optional<T> find(Class<T> type, Object primaryKey) {
         Objects.requireNonNull(type, TYPE_MUST_NOT_BE_NULL_MSG);
         Objects.requireNonNull(primaryKey, "[primaryKey] argument must be not null");
+
+        Optional<T> cachedEntity = persistenceContext.getEntity(type, primaryKey);
```

Comment:
I don't think it would make sense, if entity is in cache, then it has already passed validation. Moreover, validation is more time consuming than getting from cache